libraries/extractor/src/main/java/androidx/media3/extractor/mp4/AtomParsers.java

```diff
@@ -1237,6 +1247,22 @@ private static void parseVideoSampleEntry(
       } else if (childAtomType == Atom.TYPE_av1C) {
         ExtractorUtil.checkContainerInput(mimeType == null, /* message= */ null);
         mimeType = MimeTypes.VIDEO_AV1;
+        parent.setPosition(childStartPosition + Atom.HEADER_SIZE);
+        if (childAtomSize > Atom.HEADER_SIZE) {
+          parent.skipBytes(1); ;  // marker(1), version(7)
+          int byte2 = parent.readUnsignedByte();
+          int seqProfile = byte2 >> 5;
+          int byte3 = parent.readUnsignedByte();
+          boolean highBitdepth = ((byte3 >> 6) & 0b1) != 0;
```

Comment:
The twelveBit boolean condition is only needed if (seqProfile == 2 && highBitdepth). If you could move this assignment into that segment.

Reply:
The AV1 spec is mentioned at line 1258 to describe how to set `bitdepthLuma` from `seqProfile` and `highBitdepth`, however these two are read from `AV1CodecConfigurationRecord` (from https://aomediacodec.github.io/av1-isobmff/#av1codecconfigurationbox-syntax) which looks like this:
```
aligned (8) class AV1CodecConfigurationRecord {
  unsigned int (1) marker = 1;
  unsigned int (7) version = 1;
  unsigned int (3) seq_profile;
  unsigned int (5) seq_level_idx_0;
  unsigned int (1) seq_tier_0;
  unsigned int (1) high_bitdepth;
  unsigned int (1) twelve_bit;
  unsigned int (1) monochrome;
  unsigned int (1) chroma_subsampling_x;
  unsigned int (1) chroma_subsampling_y;
  unsigned int (2) chroma_sample_position;
  unsigned int (3) reserved = 0;

  unsigned int (1) initial_presentation_delay_present;
  if (initial_presentation_delay_present) {
    unsigned int (4) initial_presentation_delay_minus_one;
  } else {
    unsigned int (4) reserved = 0;
  }

  unsigned int (8) configOBUs[];
}
```
so I think the added lines here are correct. I could still add the link to the AV1CodecConfigurationRecord spec.

Reply:
Sorry, misunderstood your comment. Yes, twelveBit can be set before is needed.

Reply:
Done